### ISSUE TITLE
Broken when only asking for `identify` scope because team and user info require `read` scope

### ISSUE THREAD

**Issue by bgentry** [kmrshntr/omniauth-slack#17]
I tried upgrading an app from version 1.0.1 to the latest version of this gem, and it breaks when trying to read attributes from the `user_info` and `team_info` hashes. That's because we're only requesting the `identify` scope from Slack, and these other hashes are only available with the `read` scope.

Here are the values of those hashes when printed at the top of the info block:
 
```ruby
# RAW INFO:
{"ok"=>true, "url"=>"https://example.slack.com/", "team"=>"Example", "user"=>"blake", "team_id"=>"TXXXXXX", "user_id"=>"UXXXXXXX"}

# TEAM INFO:
{"ok"=>false, "error"=>"missing_scope", "needed"=>"read", "provided"=>"identify"}

# USER INFO:
{"ok"=>false, "error"=>"missing_scope", "needed"=>"read", "provided"=>"identify"}
```

So when you try to call `user_info['user'].fetch('profile') ...`, it blows up due to the fact that there is no `user` attribute:

```
NoMethodError (undefined method `fetch' for nil:NilClass):
```

Likewise, the call to `team_info['team']['domain']` blows up because there is no `team` attribute.

There are a few ways this could be solved, but you probably know a lot more about the different ways this gem is being used. One thing to keep in mind is that we'd love to avoid making additional calls like this to the Slack API when they will just fail, so the best solution would avoid asking for info that can't be fetched. 

Also, the `team_domain` seems like it could just use the `url` attribute of the `raw_info` hash rather than depending on the `team_info` hash. Unless those are sometimes different?

I likely don't have the time to solve this right now, but figured I'd at least give you a nice bug report :)

I'm happy to test out a possible solution if anybody gets around to building one.

cc @hassox @visnup @nolman for reference

**Comment by ssaunier** [comment]
I stumbled open the same issue. Sticking to 1.0.1 waiting a fix which let us use the `identify` scope. Thanks!

**Comment by kmrshntr** [comment]
Just released v2.1.2, which allows missing user_info or team_info and works with the identify scope.
